go/client/cmd_audit.go

```diff
@@ -0,0 +1,221 @@
+	}
+}
+
+type CmdAuditBox struct {
+	libkb.Contextified
+	AuditAllKnownTeams  bool
+	IsInJail            bool
+	Audit               bool
+	Attempt             bool
+	RotateBeforeAttempt bool
+	Ls                  bool
+	TeamID              keybase1.TeamID
+}
+
+func NewCmdAuditBox(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	cmd := &CmdAuditBox{
+		Contextified: libkb.NewContextified(g),
+	}
+	return cli.Command{
+		Name: "box",
+		Usage: `A team box audit makes sure a team's secrets are encrypted for
+	the right members in the team, and that when members revoke devices or
+	reset their accounts, the team's secret keys are rotated accordingly.`,
+		Flags: []cli.Flag{
+			cli.BoolFlag{
+				Name: "audit-all-known-teams",
+				Usage: `Audit all known teams. If an audit fails, the team will
+	be rotated and the audit will be retried immediately . If it fails again,
+	this error will be reported. This operation may take a long time if you are
+	in many teams.`,
+			},
+			cli.StringFlag{
+				Name:  "team-id",
+				Usage: "Team ID, required for all operations except list-known-team-ids",
+			},
+			cli.BoolFlag{
+				Name:  "is-in-jail",
+				Usage: "Check if a team id is in the box audit jail",
+			},
+			cli.BoolFlag{
+				Name:  "audit",
+				Usage: "Audit a team id, storing result to disk and scheduling additional background reaudits if it failed",
+			},
+			cli.BoolFlag{
+				Name:  "attempt",
+				Usage: "Audit a team id without persisting results anywhere",
+			},
+			cli.BoolFlag{
+				Name:  "rotate-before-attempt",
+				Usage: "Only valid with --attempt; rotate the team's keys first when given.",
+			},
+			cli.BoolFlag{
+				Name:  "list-known-team-ids",
+				Usage: "List all known team ids",
+			},
+		},
+		ArgumentHelp: "",
+		Action: func(c *cli.Context) {
+			cl.ChooseCommand(cmd, "box", c)
+		},
+	}
+}
+
+func b2i(x bool) int {
+	if x {
+		return 1
+	}
+	return 0
+}
+
+func (c *CmdAuditBox) ParseArgv(ctx *cli.Context) error {
+	c.AuditAllKnownTeams = ctx.Bool("audit-all-known-teams")
+	c.IsInJail = ctx.Bool("is-in-jail")
+	c.Audit = ctx.Bool("audit")
+	c.Attempt = ctx.Bool("attempt")
+	c.Ls = ctx.Bool("list-known-team-ids")
+	if b2i(c.AuditAllKnownTeams)+b2i(c.IsInJail)+b2i(c.Audit)+b2i(c.Attempt)+b2i(c.Ls) != 1 {
+		return fmt.Errorf("need a single command: audit-all-known-teams, is-in-jail, audit, attempt, or list-known-team-ids")
+	}
+	c.RotateBeforeAttempt = ctx.Bool("rotate-before-attempt")
+	if c.RotateBeforeAttempt && !c.Attempt {
+		return fmt.Errorf("can only use --rotate-before-attempt with --attempt")
+	}
+	c.TeamID = keybase1.TeamID(ctx.String("team-id"))
+	if c.Ls || c.AuditAllKnownTeams {
+		if len(c.TeamID) != 0 {
+			return fmt.Errorf("cannot provide team id with this option")
+		}
+	} else {
+		if len(c.TeamID) == 0 {
+			return fmt.Errorf("need team id")
+		}
+	}
+	return nil
+}
+
+type AuditResult struct {
+	teamID keybase1.TeamID
+	err    error
+}
+
+func (c *CmdAuditBox) Run() error {
```

Comment:
cleaner to make one `context.Context` for this whole function, rather than lots of `context.Background`s